### PR TITLE
URL: Fix code drift in the Editor package by removing duplicate cleanForSlug function

### PR DESCRIPTION
**Pull request by gwwar**
I was investigating reports that the object replacement character could be added to a post slug, and noticed that we had two definitions of `cleanForSlug`, with diverging behavior.

I fixed this by updating the function in the url package to include fixes from https://github.com/WordPress/gutenberg/issues/24710, and re-exporting the function in the editor package. I think we can probably deprecate the url utils in the editor package (in either a follow up or in this PR), but let me know if folks have opinions on this.

Note that this is client-side sanitization, so this will help prevent new bad entries. For posts/slugs with existing bad characters we'll need a separate patch for server sanitization.

cc @claudiulodro @Robertght would it be possible to see if this also helps with: https://github.com/WordPress/gutenberg/issues/38637

### Testing Instructions

In console:
```js
wp.url.cleanForSlug('Καλημέρα Κόσμε');
```
returns `'καλημέρα-κόσμε'` instead of an empty string.

```js
wp.editor.cleanForSlug('Καλημέρα Κόσμε');
```
Has the same result, but also includes a deprecation warning.

CleanShot 2022-02-23 at 10 41 12@2x

- In the post editor, make sure permalinks are set to include the post title
- Create a new post and save a draft
- Try editing the permalink slug to include non-latin characters, and unicode characters like the object replacement characters
- On blur unexpected characters should be sanitized.

### CODE REVIEW

File: packages/editor/src/utils/url.js
Comment (dmsnell) [inline packages/editor/src/utils/url.js]: is this normal to export from another package? I've not seen it before so I'm guessing it's just new to me.

we could instead leave `cleanForSlug` in but add a deprecation notice to use the one in `@wordpress/url` instead, but still call that one

```ts
import deprecated from `@wordpress/deprecated`;

export function cleanForSlug( string ) {
	deprecated(
		'editor::cleanForSlug replaced by url::cleanForSlug', {
		since: '12.6.0',
		plugin: 'Gutenberg',
		alternative: 'wp.url.cleanForSlug'
	} );

	return url.cleanForSlug( string );
}
```

I'm not sure how to properly indicate the `since` and `plugin` fields or how to know when `plugin` is appropriate vs. not having it there.

Comment (gwwar) [reply packages/editor/src/utils/url.js]: > is this normal to export from another package?

Usually only when we want to alias a 3rd party lib under something like `_unstable`. I was waiting on feedback for deprecating the entire file since `getWPAdminURL` isn't particularly useful either.

I can go ahead and add a warning/changelog note for this one though.

Comment (gwwar) [reply packages/editor/src/utils/url.js]: Updated in [ed709a3](https://github.com/WordPress/gutenberg/pull/39033/commits/ed709a34d868fcb496ffef91481c591d55d4593d)

**Comment by gwwar** [comment]
I also removed `getWPAdminURL` and updated references since I was looking at the util file. If that's out of scope for this PR, we can drop [b3846a7](https://github.com/WordPress/gutenberg/pull/39033/commits/b3846a7b89b6edbb5ac9fd7fab565637a4342a4c)

File: packages/editor/src/utils/url.js
Comment (dmsnell) [inline packages/editor/src/utils/url.js:22]: this was [exported from the `@wordpress/editor` package](https://github.com/WordPress/gutenberg/blob/7970d2921b670c7058bb130cf2783474875ffe14/packages/editor/src/index.js#L8) so I wonder if we need to deprecate it as well and leave it in for now.

Comment (gwwar) [reply packages/editor/src/utils/url.js:22]: Unless I'm misreading `getWPAdminURL` is not exported in utils/index. I also don't see it available from wp.editor. getWPAdminURL in trunk.

https://github.com/WordPress/gutenberg/blob/7970d2921b670c7058bb130cf2783474875ffe14/packages/editor/src/utils/index.js#L6-L8

Comment (dmsnell) [reply packages/editor/src/utils/url.js:22]: ah very good. I saw `* from 'utils'` and got that wrong

**Comment by Mamaduka** [comment]
Thank you, @gwwar. A few days ago, this caused me some confusion, but I didn't realize we had duplicated method.

Question: Currently, we have a few places in the core where we've to "force" Latin slugs. Should we introduce a new method or add the flag to `cleanForSlug`? Example: #38861.

**Comment by gwwar** [comment]
> Currently, we have a few places in the core where we've to "force" Latin slugs.

Interesting, @Mamaduka let me add some test cases. In this case, would there be any reason for URL permalink slugs (that use a post title) and named template part slugs to diverge in behavior?

**Comment by gwwar** [comment]
@Mamaduka I've added additional tests in https://github.com/WordPress/gutenberg/pull/39033/commits/d500d808e5225907e486d63b2b97380ed38b9523 unless I'm misunderstanding something, I think https://github.com/WordPress/gutenberg/pull/38861 can be updated to use wp.url.cleanForSlug instead of lodash kebabCase once this PR lands

**Review by Mamaduka — APPROVED** [review]
Thank you, @gwwar. Let's merge this, and I will handle template slug fixes in follow-up PRs.

**Review by aristath — APPROVED** [review]

**Comment by gwwar** [comment]
Excellent, thanks for the reviews folks!

**Pull request merged**
